src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,94 @@
+package games.strategy.triplea.delegate;
+
+import java.util.Collection;
+
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.BattleDelegate;
+import games.strategy.triplea.delegate.IBattle;
+import games.strategy.triplea.delegate.IBattle.BattleType;
+import games.strategy.util.Match;
+
+public class BattleDelegateOrdered extends BattleDelegate {
+
+  /**
+   * Called before the delegate will run.
+   */
+  @Override
+  public void start() {
+    super.start();
+    // Fight all air and bombing raids
+    for( final Territory t : m_battleTracker.getPendingBattleSites(true) ) {   // loop through bombing/air raids
+      final IBattle raid = m_battleTracker.getPendingBattle(t, true, null);             // Get air/bombing raid for current territory
+      if( raid == null ) {
+        try {
+          throw new Exception("Air/Bombing Raid gone missing in BattleDelegate");
```

Comment:
The intent is to document an internal failure. I could just allow a null pointer exception.